### PR TITLE
loksh: init at 6.9

### PR DESCRIPTION
**Pull request by CameronNemo**
- maintainers: add cameronnemo
- loksh: init at 6.9

###### Motivation for this change

loksh has some nice extra features that oksh does not.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x]  Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/loksh/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "loksh";
+  version = "6.9";
+
+  src = fetchFromGitHub {
+    owner = "dimkr";
+    repo = pname;
+    rev = "${version}";
```
Comment (bobby285271) [inline pkgs/shells/loksh/default.nix:17]: https://nixos.org/manual/nixpkgs/unstable/#sec-syntax

```suggestion
    rev = version;
```

Comment (CameronNemo) [reply pkgs/shells/loksh/default.nix]: thanks for the suggestion. made the change.

File: pkgs/shells/loksh/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    license = licenses.publicDomain;
+    maintainers = with maintainers; [ cameronnemo ];
+    platforms = platforms.linux;
+  };
+
+  passthru = {
+    shellPath = "/bin/loksh";
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/shells/loksh/default.nix:50]:
```suggestion
  };
}
```

File: pkgs/shells/loksh/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  buildInputs = [
+    ncurses
+  ];
+
+  postInstall = ''
+    mv $out/bin/ksh $out/bin/loksh
+    mv $out/share/man/man1/ksh.1 $out/share/man/man1/loksh.1
+    mv $out/share/man/man1/sh.1 $out/share/man/man1/loksh-sh.1
+  '';
```
Comment (SuperSandro2000) [inline pkgs/shells/loksh/default.nix:36]:
```suggestion
  '';

  passthru = {
    shellPath = "/bin/loksh";
  };
```

Comment (CameronNemo) [reply pkgs/shells/loksh/default.nix:36]: @SuperSandro2000 How do I know which order to put the attributes in generally? I was just working off the `oksh` derivation in this case.

Comment (bobby285271) [reply pkgs/shells/loksh/default.nix:36]: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md

**Pull request merged**